torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -257,8 +257,14 @@ void PropagateCatShape(Node * cat_node) {
     }
     return false;
   };
-  auto list_node = cat_node->namedInput(attr::tensors)->node();
-  if (list_node->kind() == prim::ListConstruct) {
+  if (cat_node->kind() == prim::FusedConcat) {
+    auto tensors = cat_node->inputs();
+  }
+  auto list_node = ((cat_node->kind() == prim::FusedConcat)
+		    ? cat_node
+                    : cat_node->namedInput(attr::tensors)->node());
+  if (list_node->kind() == prim::ListConstruct
+      || cat_node->kind() == prim::FusedConcat) {
```

Comment:
IMO, would be slightly more readable if it's `list_node` here.

Reply:
While I agree that `list_node` is more readable, I wonder if `cat_node` is more precise here, otherwise we rely on `list_node` not being `prim::FusedConcat` if `cat_node` is `aten::cat` (which certainly should be true, but isn't locally implied by the code) because that case isn't intended to be captured.